### PR TITLE
OCSP: fix memory leak in OCSP_url_svcloc_new method

### LINKED ISSUE #8949: Memory leak in OCSP_url_svcloc_new
Update: the original issue described was not a leak but rather confusion about whether an `ACCESS_DESCRIPTION *` contains an empty `GENERAL_NAME *` already. However, OpenSSL itself leaks a `GENERAL_NAME *`, as you can see in the second comment

### PR DESCRIPTION
**Pull request by FdaSilvaYY**
Add a few coverage test case.

Fixes #8949 

##### Checklist
- [ ] documentation is added or updated
- [X] tests are added or updated

### CODE REVIEW

**Comment by FdaSilvaYY** [comment]
this leak is very old AFAI see.
So this fix targets all active branch.
Waiting for CI pass.

File: crypto/ocsp/ocsp_ext.c [hunk 1]
```diff
@@ -449,12 +450,11 @@ X509_EXTENSION *OCSP_url_svcloc_new(X509_NAME *issuer, const char **urls)
             goto err;
         if ((ad->method = OBJ_nid2obj(NID_ad_OCSP)) == NULL)
             goto err;
-        if ((ad->location = GENERAL_NAME_new()) == NULL)
-            goto err;
         if ((ia5 = ASN1_IA5STRING_new()) == NULL)
             goto err;
         if (!ASN1_STRING_set((ASN1_STRING *)ia5, *urls, -1))
             goto err;
+        /* ad->location is init trougth ASN1_SIMPLE mechanism */
```
Comment (mattcaswell) [inline crypto/ocsp/ocsp_ext.c]: Typo: through

Although I'm not entirely sure what you mean by "through ASN1_SIMPLE" mechanism. Presumably you mean it is created as part of `ACCESS_DESCRIPTION_new()`

File: test/ocspapitest.c [hunk 2]
```diff
@@ -131,7 +149,64 @@ static int test_resp_signer(void)
...
+static int test_access_description(int testcase)
+{
+    ACCESS_DESCRIPTION *ad = ACCESS_DESCRIPTION_new();
+    int ret = 0;
+
+    if (!TEST_ptr(ad))
+        goto err;
+
+    switch (testcase)
+    {
```
Comment (mattcaswell) [inline test/ocspapitest.c]: Style nit: { should be on same line as switch

File: crypto/ocsp/ocsp_ext.c (hunk 1 above)
Comment (FdaSilvaYY) [reply crypto/ocsp/ocsp_ext.c]: AFAI see : 
- any asn1 construct 
- who has field(s) declared using ASN1_SIMPLE
- field(s) that are non-POD type.
then,  asn1 mechanism is allocating them .
The leak detection tool says it happens in `tsan1_new.c`

I ask myself if this is a (new) feature or a very old bug ?

**Comment by FdaSilvaYY** [comment]
ping @openssl : there were two missing `break;` in my test code. that i just fix and re-push.

**Comment by paulidale** [comment]
Merged to 1.1.1 and master, thanks for the contribution.

**Pull request closed without merging**